### DISCUSSION TITLE
Does Turborepo remote cache store a different copy per "origin" operating system?

### DISCUSSION THREAD

**Discussion by castarco** [vercel/turborepo#6029] [Help] [answered]
Hi.

I wanted to ask if Turborepo makes a distinction between cached results coming from different operating systems (which might be relevant when it comes to generated artifacts, or results).

For example, some tests might be OK in Linux, but fail miserably in Mac. I'm asking in part because I have some pipelines for an opensource project and I noticed that they are noticeably slower in Mac and Windows, but quite fast in Linux (and part of that slowness is that there are more cache misses in Windows and Mac).

**Comment by mehulkar** [comment, marked as answer]
Turborepo does not partition cache by OS by default, but one possible solution would be to use the [`env`](https://turbo.build/repo/docs/reference/configuration#env) or [`globalEnv`](https://turbo.build/repo/docs/reference/configuration#globalenv) configs to partition based on an environment variable like [`OSTYPE`](https://megamorf.gitlab.io/2021/05/08/detect-operating-system-in-shell-script/)
